planner/core/rule_aggregation_elimination.go

```diff
@@ -136,6 +145,114 @@ func (a *aggregationEliminateChecker) wrapCastFunction(ctx sessionctx.Context, a
 	return expression.BuildCastFunction(ctx, arg, targetTp)
 }
 
+// tryToEliminateAggregationByMapping tries to eliminate an aggregation from nested aggregations.
+// Compared with tryToEliminateAggregationByUniqueKey, this rule handles 2 more types of cases:
+//
+// 1. Nested aggregations with same group-by items, while the items contains scalar functions rather than raw columns, for example:
+//  `select p, max(dt) from (select a + b as p, count(d) as dt from t group by a + b) tt group by p`
+//  can be rewritten as
+//  `select a + b as p, count(d) as `max(dt)` from t group by a + b`
+//
+// 2. Nested aggregations in which outer group-by items are proper subset of the inner ones, for example:
+//  `select at, max(bt) from (select a as at, max(b) as bt from t group by a, c) tt group by at`
+//  can be rewritten as
+//  `select a as at, max(b) as `max(bt)` from t group by a`
+//
+// In order to apply the optimization, we tries to map definition of outer aggregation to schema of inner aggregation and check if rules are met.
+func (a *aggregationEliminateChecker) tryToEliminateAggregationByMapping(la *LogicalAggregation) LogicalPlan {
+	ptn := &nestedAggPattern{outer: la, isTrivial: true}
+	collectNestedAggPattern(la, ptn)
+
+	if ptn.inner == nil || ptn.proj == nil {
+		return nil
+	}
+	exprMap, aggMap := genColMaps(ptn)
+
+	// Map group-by items in outer aggregation to the schema of inner aggregation, so the group-by items can be compared.
+	_, items := exprSubstitute(la.ctx, exprMap, ptn.outer.GroupByItems)
+	_, items = aggSubstitute(la.ctx, aggMap, items, func(fun *aggregation.AggFuncDesc) bool {
+		// Outer group-by items cannot be aggregated result of inner aggregation.
```

Comment:
The check here is to prevent the following case:

```
select bt, max(ct) from (select b as bt, count(c) as ct from t group by c, b) tt group by ct;
```
`ct` is defined as `count(c)` and it is tricky to indicate the semantics of aggregations here.